.github/workflows/apiBreakTest.yml

```diff
@@ -0,0 +1,174 @@
+          swift api-digester -sdk "$SDK_PATH" -dump-sdk -module "$module" -o "$NEW_API_DIR/${module}.json" -I .build/debug || { echo "Failed to dump new SDK for module $module"; exit 1; }
+        done
+        
+        # Compare APIs for each module and capture the output
+        api_diff_output=""
+        for module in $modules; do
+          swift api-digester -sdk "$SDK_PATH" -diagnose-sdk --input-paths "$OLD_API_DIR/${module}.json" --input-paths "$NEW_API_DIR/${module}.json" >> "api-diff-report-${module}.txt" 2>&1
+          module_diff_output=$(grep -v '^/\*' "api-diff-report-${module}.txt" | grep -v '^$' | grep -v 'has been added as a new enum case$' | grep -v 'is now with @_spi$' || true)
+          if [ -n "$module_diff_output" ]; then
+            api_diff_output="${api_diff_output}\n**Module: ${module}**\n${module_diff_output}\n"
+
+            # Check if there are lines containing "has been renamed to Func"
+            if echo "$module_diff_output" | grep -q 'has been renamed to Func'; then
+                # Capture the line containing "has been renamed to Func"
+                renamed_line=$(echo "$module_diff_output" | grep 'has been renamed to Func')
+            
+                # Append a message to the module_diff_output
+                api_diff_output="${api_diff_output}👉🏻 _Note: If you're just adding optional parameters to existing methods, neglect the line:_\n_${renamed_line}_\n"
+            fi
+          fi
+        done
+        
+        echo "API_DIFF_OUTPUT<<EOF" >> $GITHUB_ENV
+        if [ -n "$api_diff_output" ]; then
+          echo "### 💔 Public API Breaking Change detected:" >> $GITHUB_ENV
+          echo -e "$api_diff_output" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+          echo "BREAKING_CHANGE_DETECTED=true" >> $GITHUB_ENV
+        else
+          echo "### ✅ No Public API Breaking Change detected" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+          echo "BREAKING_CHANGE_DETECTED=false" >> $GITHUB_ENV
+        fi
+
+        # Checkout to the branch associated with the pull request
+        git stash --include-untracked
+        git checkout ${{ github.head_ref }}
+
+        if [ ! -d "api-dump" ]; then
+            echo "api-dump folder does not exist. Creating it..."
+            mkdir -p "api-dump"
+        fi
+        
+        # Update the api-dump folder of the new version by making a commit if there are changes
+        for module in $modules; do
+            if [ ! -f api-dump/${module}.json ]; then
+              echo "API file does not exist in api-dump folder. Creating it..."
+              echo "{}" > "api-dump/${module}.json"
+            fi
+            if ! diff "$NEW_API_DIR/${module}.json" "api-dump/${module}.json" > /dev/null; then
+                echo "Updating API Dumps..."
+                mv "$NEW_API_DIR/${module}.json" "api-dump/${module}.json"
+            fi
+        done
+
+        git config --global user.name "github-actions[bot]"
```

Comment:
can we use the amplify ops user/account here?

Reply:
Where can I get the information for the amplify ops account, like its username and email? Thanks!

Reply:
Resolved